src/web/server.rkt

```diff
@@ -0,0 +1,161 @@
+#lang racket
+
+(require openssl/sha1)
+
+(require "../sandbox.rkt" "../config.rkt" "../syntax/read.rkt")
+(require (submod "../timeline.rkt" debug))
+
+(provide completed-job? get-results-for get-improve-job-data job-count
+ is-server-up create-job start-job is-job-finished wait-for-job
+ start-job-server)
+
+#| Job Server Public API section |#
+
+; Helers to isolated *completed-jobs*
+(define (completed-job? id) 
+ (hash-has-key? *completed-jobs* id))
+
+(define (get-results-for id) 
+ (hash-ref *completed-jobs* id))
+
+; I don't like how specific this function is but it keeps the API boundary.
+(define (get-improve-job-data)
+ (for/list ([(k v) (in-hash *completed-jobs*)]
+     #:when (equal? (job-result-command v) 'improve))
+      (get-table-data v (format "~a.~a" k *herbie-commit*))))
+
+(define (job-count)
+ (hash-count *job-status*))
+
+(define (is-server-up)
+ (thread-running? *worker-thread*))
+
+;; Creates a command object to be passed to start-job server.
+;; TODO contract?
+(define (create-job command formula 
```

Comment:
This is a pretty ugly (long) signature. I think it's fine to merge, but longer term one thing we might consider is just turning on profiling and timeline for every command so we don't need to think about them.

Reply:
Suggesting we shouldn't surface the option at the call site and just make that a global setting in `config.rkt`?

Reply:
I was thinking just leave it permanently on, don't even surface an option. I think the original worry was that the overhead from timeline and profiling might be too high, but it turns out to be pretty mild.